src/app/ad-authoring/ad-authoring.state.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import {Observable, BehaviorSubject} from 'rxjs';
+import {AdAuthoring} from './ad-authoring';
+
+export interface AdAuthoringState {
+  readonly adAuthorings: ReadonlyArray<AdAuthoring>;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdAuthoringStateContainer {
+
+    private state$: BehaviorSubject<AdAuthoringState>;
+
+    protected constructor () {
+        this.state$ = new BehaviorSubject({
+          adAuthorings: [],
+        });
```

Comment:
We can inline the state assignment above where the state$ property is defined and remove the explicit constructor.